scripts/record_and_replay.py

```diff
@@ -0,0 +1,167 @@
+    except:
+        print("ERROR: could not read info from terraform")
+        sys.exit(1)
+
+
+# get resources of importance
+validators = []
+fullnodes = []
+validators_ecs = []
+fullnodes_ecs = []
+for res in out.get('values').get('root_module').get('resources'):
+    addr = res.get('address')
+    if 'aws_instance.validator' in addr:
+        validators.append(res)
+    elif 'aws_instance.fullnode' in addr:
+        fullnodes.append(res)
+    elif 'ecs_task_definition.validator' in addr:
+        validators_ecs.append(res)
+    elif 'ecs_task_definition.fullnode' in addr:
+        fullnodes_ecs.append(res)
+
+
+# terraform show not guaranteed to be sorted
+list.sort(validators, key=lambda x: x.get('index'))
+list.sort(fullnodes, key=lambda x: x.get('index'))
+list.sort(validators_ecs, key=lambda x: x.get('index'))
+list.sort(fullnodes_ecs, key=lambda x: x.get('index'))
+
+
+# parse validators
+validator_ips = []
+validator_restore_vols = []
+for validator in validators:
+    vals = validator.get('values')
+    validator_ips.append(vals.get('private_ip'))
+    for vol in vals.get('ebs_block_device'):
+        if vol.get('device_name') == '/dev/xvdb':
+            validator_restore_vols.append(vol.get('volume_id'))
+
+
+# parse fullnodes
+fullnode_ips = []
+for fullnode in fullnodes:
+    fullnode_ips.append(fullnode.get('values').get('private_ip'))
+
+
+# Get image and version for ECS
+logstash_image = None
+validator_image = None
+safetyrules_image = None
+logstash_version = None
+validator_versions = []
+safetyrules_version = None
+for ecs in validators_ecs:
+    containers = json.loads(ecs.get('values').get('container_definitions'))
+    for container in containers:
+        container_name = container.get('name')
+        try:
+            image, version = container.get('image').split(':')
+        except:
+            print("Error getting ECS images")
+            sys.exit(1)
+        if container_name == 'logstash':
+            logstash_image = image
+            logstash_version = version
+        elif container_name == 'validator':
+            validator_image = image
+            validator_versions.append(version)
+        elif container_name == 'safety-rules':
+            safetyrules_image = image
+            safetyrules_version = version
+
+# build the var strings
+vars = []
+vars.append(gen_tf_var_list('restore_vol_ids', validator_restore_vols))
+
+vars.append(gen_tf_var_list('override_validator_ips', validator_ips))
+vars.append(gen_tf_var_list('override_fullnode_ips', fullnode_ips))
+
+vars.append(gen_tf_var_str('image_repo', validator_image))
+vars.append(gen_tf_var_list('override_image_tags', validator_versions))
+vars.append(gen_tf_var_str('logstash_image', logstash_image))
+vars.append(gen_tf_var_str('logstash_version', logstash_version))
+vars.append(gen_tf_var_str('safety_rules_image_repo', safetyrules_image))
+vars.append(gen_tf_var_str('safety_rules_image_tag', safetyrules_version))
+
+if args.var_file:
```

Comment:
Why do you need this merge behaviour?

Reply:
The idea was that if you knew exactly which var file was used to generate the source network, you could use that as a basis for a more accurate replay.

Reply:
An alternative might be to not merge and have the operator do something like `terraform apply --var-file=dev.tfvars --var-file=replay.tfvars` but the generated variables should really only be used for replay of a particular network instance, hence the merging with the original var file to clarify that.

Reply:
I would suggest that approach, I think writing a new file with the replay-specific vars makes more sense than messing with existing var files which people have to remember to revert later.

Reply:
This makes a copy of an existing var file and merges with that, so nothing is overwritten and revert is not needed. Users can specify an out_file name, or it defaults to [`roar.tfvars`](https://github.com/libra/libra/pull/3716/files#diff-ba032759c4638e3f269da638887ff564R148). In either case, the original file is untouched

Reply:
Okay, your call, but I still think the merge behaviour is unnecessary and complex.